### PR TITLE
Missions screen

### PR DESCRIPTION
**Pull request by tddumont**
This PR contains an update to the task rubric, tests for the Missions screen, and an update to the Mission overview.

### CODE REVIEW

File: coverage/lcov-report/base.css [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    background: url(sort-arrow-sprite.png) no-repeat scroll 0 0 transparent;
+}
+.coverage-summary .sorted .sorter {
+    background-position: 0 -20px;
+}
+.coverage-summary .sorted-desc .sorter {
+    background-position: 0 -10px;
+}
+.status-line {  height: 10px; }
+/* yellow */
```
Comment (samiur98) [inline coverage/lcov-report/base.css:155]: We Should incorporate commenting on our CSS Documentation a lot more. Good job!

File: coverage/lcov-report/block-navigation.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    // Classes of code we would like to highlight in the file view
+    var missingCoverageClasses = ['.cbranch-no', '.cstat-no', '.fstat-no'];
+
+    // Elements to highlight in the file listing view
+    var fileListingElements = ['td.pct.low'];
+
+    // We don't want to select elements that are direct descendants of another match
+    var notSelector = ':not(' + missingCoverageClasses.join('):not(') + ') > '; // becomes `:not(a):not(b) > `
+
+    // Selecter that finds elements on the page to which we can jump
```
Comment (samiur98) [inline coverage/lcov-report/block-navigation.js:12]: Looks Good.

File: src/Components/MissionsScreen.js [hunk 3]
```diff
@@ -1,95 +1,136 @@
...
+    const { loading: progressLoading, error: progressError, data: progressData, refetch : progressRefetch} = useQuery(GET_ALL_MISSION_PROGRESS, {
+      variables: { id: " " },
+    });
+
+    if(loading || progressLoading) return (
+      <div className = 'tasks'> 
+        <h1>Loading...</h1>
+      </div>
+    )
+
```
Comment (samiur98) [inline src/Components/MissionsScreen.js:31]: Looks good, thanks for commenting the code.

File: src/Components/MissionsScreen.js [hunk 4]
```diff
@@ -1,95 +1,136 @@
...
-        console.log(data);
-        return data.missions.map((mission) => (
-        <div className="Mission" onClick={()=>setFocusedMission(mission)}>
-          <HdrWeakIcon style={{ color: "white", transform: "scale(6)" }}/>
-          {/* <h1 style={{size: "10px", color: "white"}}>{mission.name}</h1> */}
-        </div>));
+    const { loading: progressLoading, error: progressError, data: progressData, refetch : progressRefetch} = useQuery(GET_ALL_MISSION_PROGRESS, {
+      variables: { id: " " },
+    });
+
```
Comment (ConnorBoulais) [inline src/Components/MissionsScreen.js:25]: Great job hooking up to the backend, hopefully this API is fully implemented soon

File: src/Components/MissionsScreen.js [hunk 5]
```diff
@@ -1,95 +1,136 @@
...
+//this function allows for dependency injection during testing
+export function MissionsScreenDisplay(data, progressData, hist, focusedMission, setFocusedMission) {
+
+  function displayMissionList(){
+      return (<div className="missionList"> 
+        {data.missions.map((mission) => (
+          <div data-testid={mission.id} className="Mission" onClick={()=>setFocusedMission(mission)}>
+            {/* <HdrWeakIcon style={{ color: "white", transform: "scale(6)" }}/> */}
+            <h1 style={{size: "10px", color: "white"}}>{mission.name}</h1>
```
Comment (samiur98) [inline src/Components/MissionsScreen.js:53]: Might want to get rid of line 52, does not seem to serve any purpose.

File: src/Tests/MissionsScreen.test.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+const mockFocusedMission = null;
+
+const mockSetFocusedMission = jest.fn();
+
+afterEach(() => {
+    cleanup();
+});
+
+test("Test for the course title", () => {
+    render(MissionsScreenDisplay(mockMissionData, mockProgressData, [], mockFocusedMission, mockSetFocusedMission));
```
Comment (samiur98) [inline src/Tests/MissionsScreen.test.js:59]: Good Tests.

File: src/gqlQueries.js [hunk 7]
```diff
@@ -278,6 +276,25 @@ export const GET_TASK_PROGRESS = gql`
     }
   }
 `;
+
+export const GET_ALL_MISSION_PROGRESS = gql`
+  query getMissionsProgress($id: String!) {
+    getAllMissionProgress(courseId: $id) {
+      mission {
+        id
+        name
```
Comment (samiur98) [inline src/gqlQueries.js:285]: Good job making the query

File: src/Components/NavDrawer.js [hunk 8]
```diff
@@ -65,18 +63,13 @@ function NavDrawer(props) {
...
             <ListItemText className="buttonText" primary="Missions" />
           </ListItem>
 
-          
-          <ListItem button key="Task" onClick={() => (props.history.push('/task'))}>
-            <ListItemIcon><LinearScaleIcon /></ListItemIcon>
-            <ListItemText className="buttonText" primary="Task" />
-          </ListItem>
-
-          <ListItem button key="Courses" onClick={() => (props.history.push('/courses'))}>
```
Comment (ConnorBoulais) [inline src/Components/NavDrawer.js:79]: Awesome, this caused multiple defect reports

**Review by samiur98 — APPROVED** [review]
Minor issues, but ready to be merged in.

File: src/Components/Task.js [hunk 9]
```diff
@@ -295,7 +295,13 @@ function Task(props) {
...
 
   return (
-    <div className = 'tasks'>  
+    <div className = 'tasks' style={rubricOpen? {
+      width: `calc(100% - ${500}px)`,
+      marginRight: "500px", 
+      } : {
+        width: `calc(100% - ${0}px)`,
+        marginRight: "0px",
+        }}>
```
Comment (ConnorBoulais) [inline src/Components/Task.js:304]: Sweet, this is what Christian wanted

File: src/Tests/MissionsScreen.test.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
+import { render, screen, cleanup, fireEvent } from '@testing-library/react';
+import {MissionsScreenDisplay, calculateMissionProgress, redirectToMission} from '../Components/MissionsScreen.js';
+
+const mockProgressData = {
+      "getAllMissionProgress": [
+        {
+          "mission": {
+            "id": "MISSION#123",
```
Comment (ConnorBoulais) [inline src/Tests/MissionsScreen.test.js:8]: nice tests

**Review by ConnorBoulais — APPROVED** [review]

**Pull request merged**